pytest_factoryboy/fixture.py

```diff
@@ -1,51 +1,103 @@
 """Factory boy fixture integration."""
+from __future__ import annotations
 
+import atexit
+import pathlib
+import shutil
 import sys
+import importlib.util
+import tempfile
+import typing
+from dataclasses import dataclass, field
+from functools import lru_cache
+from inspect import getmodule, signature
 
 import factory
 import factory.builder
 import factory.declarations
 import factory.enums
 import inflection
-import pytest
-
-from inspect import getmodule, signature
+import mako.template
 
 from pytest_factoryboy.compat import PostGenerationContext
 
+
 SEPARATOR = "__"
 
 
-FIXTURE_FUNC_FORMAT = """
-def {name}({deps}):
-    return _fixture_impl(request, **kwargs)
+module_template = mako.template.Template(
+    """\
+import pytest
+from pytest_factoryboy.fixture import (
+    attr_fixture,
+    factory_fixture,
+    model_fixture,
+    subfactory_fixture,
+)
+
+__all__ = (
+% for fixture_def in fixture_defs:
+    ${ repr(fixture_def.name)},
+% endfor
+)
+
+
+def _fixture(related):
+    def fixture_maker(fn):
+        fn._factoryboy_related = related
+        return pytest.fixture(fn)
+
+    return fixture_maker
+% for fixture_def in fixture_defs:
+
+
+${ fixture_def.kwargs_var_name } = {}
+
+
+@_fixture(related=${ repr(fixture_def.related) })
+def ${ fixture_def.name }(
```

Comment:
NIP: now, we could even add simple docstrings to fixtures. Something like:

+ `my_app.MyModel instances factory.`
+ `Factory of MyModelFactory.my_attribute attribute.`

Docstrings generation can be implemented on `FixtureDef` dataclass.

Reply:
that could be a good idea, but I don't feel like thinking about the docstring at the moment, maybe an improvement for a next iteration